docs-guides/source/opt-stable-diffusion.md

```diff
@@ -0,0 +1,188 @@
+For the purpose of this tutorial we will use a fixed prompt - _“cat in a tuxedo, oil on canvas”_, to compare output images produced by different model variants [^1]. 
+
+Execute the following command to generate image from the baseline SDXL pipeline using CoreML models generated above:
+```shell      
+python -m python_coreml_stable_diffusion.pipeline --prompt "cat in a tuxedo, oil on canvas" \
+-i output-xl-512 -o base_xl --compute-unit CPU_ONLY --seed 231 \
+--model-version stabilityai/stable-diffusion-xl-base-1.0
+```
+
+<img height="250px" src="images/sdxl-base.png" width="250px"/>
+
+[^1]: All images shown in this article have been generated on macOS 15.0. The image outputs can vary based on platform, backend, number of inference steps, etc.
+
+## Palettizing CoreML Model
+            
+### Scalar palettization
+Let's start with data free scalar palettization using [ct.optimize.coreml.palettize_weights](https://apple.github.io/coremltools/source/coremltools.optimize.coreml.post_training_quantization.html#coremltools.optimize.coreml.palettize_weights) API. 
+
+```python
+import coremltools as ct
+import coremltools.optimize as cto
+
+model = ct.models.MLModel(
+    "output-xl-512/Stable_Diffusion_version_stabilityai_stable-diffusion-xl-base-1.0_unet.mlpackage"
+)
+config = cto.coreml.OptimizationConfig(
+    global_config=cto.coreml.OpPalettizerConfig(nbits=4),
+)
+compressed_model = cto.coreml.palettize_weights(model, config)
+```
+    
+Trying with nbits equal to 8, 6 & 4 bits we see the following results: 
+
+| Config                | Model Size | Image                                                           |
+|-----------------------|------------|-----------------------------------------------------------------|
+| 8-bit (per tensor)    | 2.40 GB    | <img height="250px" src="images/sdxl-8bit.png" width="250px"/>  |
+| 6-bit (per-tensor)    | 1.80 GB    | <img height="250px" src="images/sdxl-6bit.png" width="250px"/>  |
+| 4-bit (per-tensor)    | 1.21 GB    | <img height="250px" src="images/sdxl-4bit.png" width="250px"/>  | 
+
+Applying 8bit palettization can reduce the model size to be about half of the `float16` model, but it is still quite large to consider iOS integration. 
+With 6bit we can finally run this model on an iPad. With 4bit compression we are unable to get a good image anymore, meaning the model isn’t accurate. 
+             
+### Grouped channel palettization
+Let's try to regain the accuracy loss from 4 bit palettization by applying the [per_grouped_channel](opt-palettization-overview.md#granularity) palettization, which increases the number of LUTs (look up tables) per weight tensor.
+
+```python
+import coremltools as ct
+import coremltools.optimize as cto
+
+mlmodel = ct.models.MLModel(
+    "output-xl-512/Stable_Diffusion_version_stabilityai_stable-diffusion-xl-base-1.0_unet.mlpackage"
+)
+op_config = cto.coreml.OpPalettizerConfig(
```

Comment:
I think it's totally reasonable to assume, the code in the previous code block has been executed. I would delete all the code in this code block before this line.